client/src/components/ViewPostDialog.tsx

```diff
@@ -472,31 +471,14 @@ export default function ViewPostDialog() {
         console.error(`Error fetching post ${err}`);
         toggleError(true);
       });
-  };
-
-  React.useEffect(() => {
-    // Fetch data (on initial load)
-    if (!postData.User && !error) {
-      fetchData();
-    }
-  });
+  }, [postid, userContext.data]);
 
   React.useEffect(() => {
-    if (!isEditing && !error) {
-      // ensure data updates instantly for the user that finished editing
-      fetchData();
-    }
-  }, [isEditing]);
-
-  React.useEffect(() => {
-    /* Fetch incase data has changed / post was edited */
     if (!error) {
-      const interval = setInterval(() => {
-        fetchData();
-      }, 500);
-      return () => clearInterval(interval);
+      fetchData();
```

Comment:
Do not fetch data if the user `isEditing`

Reply:
its fine, if anything it will ensure that they get the most recent update to the post before they make changes to it

Reply:
This might cause weird behavior because the input boxes have default value of what the postdata is, but if we keep fetching data, it might bug it out. Also, if we fetch a post at the same time that the user clicks 'update' (after finishing editing) there will be lag before the request is sent.